### ISSUE TITLE
MSF and type SCT annotations

### ISSUE THREAD

**Issue by sarranz** [jasmin-lang/jasmin#962]
This gives an error
```
fn extra_msf_swap(#[msf, public] reg u64 x) {}
```
and it makes sense. But this doesn't
```
fn extra_msf_poly(#[poly=t, msf] reg u64 x) {}
```
Is this on purpose?

**Comment by vbgl** [comment]
No. Not on purpose. The error comes from the fact that `public` and `msf` are incompatible annotations. Do you think they should be allowed both?

**Comment by sarranz** [comment]
I don't think it makes a big difference either way. I initially expected both to be rejected. Do we get any benefit from specifying the confidentiality type of an MSF?
